src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -798,201 +915,183 @@ def get_traces(self, start_frame, end_frame, channel_indices):
-                else:
-                    raise ValueError(
-                        "Multiple ElectricalSeries found in the file. "
-                        "Please specify the 'electrical_series_name' argument:"
-                        f"Available options are: {available_electrical_series}."
-                    )
-            else:
-                if electrical_series_name not in available_electrical_series:
-                    raise ValueError(
-                        f"'{electrical_series_name}' not found in the file. "
-                        f"Available options are: {available_electrical_series}"
-                    )
-            self.electrical_series_location = available_electrical_series[electrical_series_name]
-            electrical_series = hdf5_file[self.electrical_series_location]
+        When a `t_start` is not provided it will be inferred from the corresponding ElectricalSeries with name equal
+        to `electrical_series_path`. The `t_start` then will be either the `ElectricalSeries.starting_time` or the
+        first timestamp in the `ElectricalSeries.timestamps`.
+    storage_options: dict | None = None,
+        Additional parameters for the storage backend (e.g. AWS credentials) used for "zarr" stream_mode.
+    use_pynwb: bool, default: False
+        Uses the pynwb library to read the NWB file. Setting this to False, the default, uses h5py
+        to read the file. Using h5py can improve performance by bypassing some of the PyNWB validations.
 
-            # Get sampling frequency
-            if "starting_time" in electrical_series.keys():
-                t_start = electrical_series["starting_time"][()]
-                sampling_frequency = electrical_series["starting_time"].attrs["rate"]
-            elif "timestamps" in electrical_series.keys():
-                timestamps = electrical_series["timestamps"][:]
-                t_start = timestamps[0]
-                sampling_frequency = 1.0 / np.median(np.diff(timestamps[:samples_for_rate_estimation]))
+    Returns
+    -------
+    sorting: NwbSortingExtractor
+        The sorting extractor for the NWB file.
+    """
 
-            self.t_start = t_start
+    extractor_name = "NwbSorting"
+    mode = "file"
+    installation_mesg = "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+    name = "nwb"
 
-        assert (
-            sampling_frequency is not None
-        ), "Couldn't load sampling frequency. Please provide it with the 'sampling_frequency' argument"
-        assert (
-            self.t_start is not None
-        ), "Couldn't load a starting time for the sorting. Please provide it with the 't_start' argument"
+    def __init__(
+        self,
+        file_path: str | Path,
+        electrical_series_path: str | None = None,
+        sampling_frequency: float | None = None,
+        samples_for_rate_estimation: int = 1_000,
+        unit_table_path: str = "units",
```

Comment:
I would like to have these path after the *; to make it keyword only but do I remember well from the discussion with Cody that you don't like them?